util/file_util.py

```diff
@@ -0,0 +1,922 @@
+      filename: name of csv file to be copied into the spreadsheet.
+      url: Url for the spreadsheet into which file is copied.
+      worksheet: worksheet title into whcih file is copied.
+        If not set, the file is copied into the first sheet.
+    Returns:
+      the url for the worksheet into whcih file was copied.
+    '''
+    # Read the rows from the source file
+    rows = []
+    with FileIO(filename) as file:
+        csv_reader = csv.reader(file, skipinitialspace=True, escapechar='\\')
+        for row in csv_reader:
+            rows.append(row)
+
+    logging.debug(f'Writing {len(rows)} rows from {filename} into {url}')
+    # Get the worksheet handle.
+    ws = file_get_gspread_worksheet(url, worksheet)
+    if not ws:
+        logging.error(f'Unable to get worksheet {url}, {worksheet}')
+        return ''
+
+    # Clear the worksheet
+    ws.clear()
+    # Add all the rows.
+    ws.update(rows)
+    logging.debug(
+        f'Wrote {len(rows)} rows from {filename} into spreadsheet:{url},{ws.title}'
+    )
+    return ws.url
+
+
+def file_is_csv(filename: str) -> bool:
+    '''Returns True is the file has a .csv extension or is a spreadsheet.'''
+    if filename.endswith('.csv') or file_is_google_spreadsheet(filename):
+        return True
+    return False
+
+
+def _copy_file_chunks(src, dst, chunk_size=1000000):
+    '''Copy file content from src to dst in chunks of given size.'''
+    buf = src.read(chunk_size)
+    while (len(buf)):
+        dst.write(buf)
+        buf = src.read(chunk_size)
+
+
+def _aggregate_value(value1: str, value2: str, aggregate: str = 'sum') -> str:
+    '''Return value aggregated from value1 and value2 as per the aggregate setting.
+    Args:
+      value1: value to be aggregated from source
+      value2: value to be aggregated into from destination
+      aggregate: string setting for aggregation method which is one of
+        sum, min, max, list
+    Returns:
+      aggregated value
+    '''
+    value = None
+    if isinstance(value1, str) or isinstance(value2, str):
+        if aggregate == 'sum':
+            # Use list for combining string values.
+            aggregate = 'list'
+    if isinstance(value1, set) or isinstance(value2, set):
+        aggregate = 'set'
+    if aggregate == 'sum':
+        value = value1 + value2
+    elif aggregate == 'min':
+        value = min(value1, value2)
+    elif aggregate == 'max':
+        value = max(value1, value2)
+    elif aggregate == 'list':
+        # Create a comma separated list of unique values combining lists.
+        value = set(str(value1).split(','))
+        value.update(str(value2).split(','))
+        value = ','.join(sorted(value))
+    elif aggregate == 'set':
+        value = set(value1)
+        value.update(value2)
+    else:
+        logging.fatal(
+            f'Unsupported aggregation: {aggregate} for {value1}, {value2}')
+    return value
+
+
+def _dict_aggregate_values(src: dict, dst: dict, config: dict) -> dict:
```

Comment:
Should stuff like aggregation be in file_util? :)

Reply:
They were all in scripts/statvar/util.py.

Can I create a string_util or a dict_util for this?

Reply:
aggregation_util ?

Reply:
added aggregation_util